src/toast/ops/sim_catalog.py

```diff
@@ -0,0 +1,577 @@
+                sigma = fwhm * gaussian_fwhm_to_sigma
+                w = 2 * fwhm
+                n = 101  # Should be odd to include origin
+                x = np.linspace(-w, w, n)
+                y = np.linspace(-w, w, n)
+                X, Y = np.meshgrid(x, y)
+                model = np.exp(-(X**2 + Y**2) / (2 * sigma**2)).to_value()
+                beam_dict = {
+                    "data": model,
+                    "size": 2 * w,
+                    "npix": n,
+                    "res": 2 * w / (n - 1),
+                }
+            else:
+                with h5py.File(self.beam_file, "r") as f:
+                    beam_dict = {}
+                    beam_dict["data"] = f["beam"][:]
+                    beam_dict["size"] = f["beam"].attrs["size"] * u.degree
+                    beam_dict["res"] = f["beam"].attrs["res"] * u.degree
+                    beam_dict["npix"] = f["beam"].attrs["npix"]
+                    self.beam_props["ALL"] = beam_dict
+
+        model = beam_dict["data"].copy()
+        model /= np.amax(model)
+
+        # DEBUG begin
+        # These commands add a tail to the beam that points towards the horizon
+        # nx, ny = np.shape(model)
+        # nhalf = nx // 2
+        # w = 10
+        # model[nhalf - w : nhalf + w + 1, 0 : nhalf] = 1
+        # DEBUG end
+
+        w = beam_dict["size"].to_value(u.rad) / 2
+        n = beam_dict["npix"]
+        x = np.linspace(-w, w, n)
+        beam = RectBivariateSpline(x, x, model)
+        # Farthest distance (corner) where beam data is available
+        r = np.sqrt(w**2 + w**2)
+
+        # Measure the solid angle using the interpolator
+
+        x = np.linspace(-w, w, 10 * n + 1)
+        dx = (x[1] - x[0]) * u.rad
+        beam_solid_angle = np.sum(beam(x, x)) * dx**2
+
+        return beam, r, beam_solid_angle
+
+    @function_timer
+    def _observe_catalog(
+        self,
+        data,
+        obs,
+        prefix,
+        dets,
+        scale,
+    ):
+        """
+        Observe the catalog with each detector in tod
+        """
+        log = Logger.get()
+
+        # Get a view of the data which contains just this single
+        # observation
+        obs_data = data.select(obs_name=obs.name)
+        focalplane = obs.telescope.focalplane
+
+        times_mjd = to_MJD(obs.shared[self.times].data)
+        if self.hwp_angle in obs.shared:
+            hwp_angle = obs.shared[self.hwp_angle].data
+        else:
+            hwp_angle = None
+        beam = None
+
+        for idet, det in enumerate(dets):
+            bandpass = obs.telescope.focalplane.bandpass
+            signal = obs.detdata[self.det_data][det]
+
+            self.detector_pointing.apply(obs_data, detectors=[det])
+            try:
+                det_quat = obs_data.obs[0].detdata[self.detector_pointing.quats][det]
+            except:
+                import pdb
+
+                pdb.set_trace()
+
+            # Convert Az/El quaternion of the detector into angles
```

Comment:
Hopefully these are Ra/Dec quaternions...

Reply:
Right. Thanks the comment was stale.